astropy/coordinates/tests/test_intermediate_transformations.py

```diff
@@ -194,6 +194,116 @@ def test_cirs_to_hadec():
     assert_allclose(cirs.dec, cirs3.dec)
 
 
+def test_itrs_topo_to_altaz_with_refraction():
+
+    loc = EarthLocation(lat=0*u.deg, lon=0*u.deg, height=0*u.m)
+    usph = golden_spiral_grid(200)
+    dist = np.linspace(1., 1000.0, len(usph)) * u.au
+    icrs = ICRS(ra=usph.lon, dec=usph.lat, distance=dist)
+    altaz_frame1 = AltAz(obstime = 'J2000', location=loc)
+    altaz_frame2 = AltAz(obstime = 'J2000', location=loc, pressure=1000.0 * u.hPa,
+                         relative_humidity=0.5)
+    itrs_frame = ITRS(location=loc)
+
+    #Normal route
+    #No Refraction
+    altaz1 = icrs.transform_to(altaz_frame1)
+
+    #Refraction added
+    altaz2 = icrs.transform_to(altaz_frame2)
+
+    #Refraction removed
+    cirs_frame = CIRS(obstime = 'J2000', location=loc)
+    cirs = altaz2.transform_to(cirs_frame)
+    altaz3 = cirs.transform_to(altaz_frame1)
+
+    #Through ITRS
+    #No Refraction
+    itrs = icrs.transform_to(itrs_frame)
+    altaz11 = itrs.transform_to(altaz_frame1)
+
+    assert_allclose(altaz11.az - altaz1.az, 0*u.mas, atol=0.1*u.mas)
+    assert_allclose(altaz11.alt - altaz1.alt, 0*u.mas, atol=0.1*u.mas)
+    assert_allclose(altaz11.distance - altaz1.distance, 0*u.cm, atol=10.0*u.cm)
+
+    #Round trip
+    itrs11 = altaz11.transform_to(itrs_frame)
+
+    assert_allclose(itrs11.x, itrs.x)
+    assert_allclose(itrs11.y, itrs.y)
+    assert_allclose(itrs11.z, itrs.z)
+
+    #Refraction added
+    altaz22 = itrs.transform_to(altaz_frame2)
+
+    assert_allclose(altaz22.az - altaz2.az, 0*u.mas, atol=0.1*u.mas)
+    assert_allclose(altaz22.alt - altaz2.alt, 0*u.mas, atol=0.1*u.mas)
+    assert_allclose(altaz22.distance - altaz2.distance, 0*u.cm, atol=10.0*u.cm)
+
+    #Refraction removed
+    itrs = altaz22.transform_to(itrs_frame)
+    altaz33 = itrs.transform_to(altaz_frame1)
+
+    assert_allclose(altaz33.az - altaz3.az, 0*u.mas, atol=0.1*u.mas)
+    assert_allclose(altaz33.alt - altaz3.alt, 0*u.mas, atol=0.1*u.mas)
+    assert_allclose(altaz33.distance - altaz3.distance, 0*u.cm, atol=10.0*u.cm)
+
+
+def test_itrs_topo_to_hadec_with_refraction():
+
+    loc = EarthLocation(lat=0*u.deg, lon=0*u.deg, height=0*u.m)
```

Comment:
As above. In principle, one can have a test class in which these initializations are done in a `setup_class` method.